src/pandas_openscm/index_manipulation.py

```diff
@@ -482,7 +482,7 @@ def update_levels(
         ini = ini.remove_unused_levels()  # type: ignore
 
     levels: list[pd.Index[Any]] = list(ini.levels)
-    codes: list[list[int] | npt.NDArray[np.integer[Any]]] = list(ini.codes)
+    codes: list[npt.NDArray[np.integer[Any]]] = list(ini.codes)
```

Comment:
I'm not sure why we had `list[int]` too, but removing it is correct I think (it gets cast to an array by pandas, so no need to support the list path too as that is slow because you're casting back and forth for now reason)